### PR TITLE
fix(operator): mutate both connected and not vs

### LINKED ISSUE #827: istio-workspace-operator restart after launch "ike develop" command
##### Issue Overview

I'm trying to use ike to make some test, but I don't work.
When I launch this command "ike develop --deployment nginx --port 80 --watch --run 'docker run -it -p 80:80 nginx' --route header:end-user=alien-ike --watch" I obtain this error in istio-workspacer-operator pod and restart it.
<details>
`{"level":"info","ts":1619887917.6047382,"logger":"root","caller":"session/session_controller.go:163","msg":"Reconciling Session","type":"controller","Request.Namespace":"default","Request.Name":"jorge.martinez-wsoqj"}
{"level":"info","ts":1619887917.6047118,"logger":"root","caller":"reference/enqueue_annotations.go:254","msg":"Reference Object created","type":"enqueue","kind":"/, Kind=","namespace":"default","name":"nginx-b23a6a84-jorge.martinez-wsoqj","targets":[{"Namespace":"default","Name":"jorge.martinez-wsoqj"}]}
{"level":"info","ts":1619887917.6097875,"logger":"root","caller":"session/session_controller.go:200","msg":"Added session","type":"controller","Request.Namespace":"default","Request.Name":"jorge.martinez-wsoqj"}
{"level":"info","ts":1619887917.6099486,"logger":"root","caller":"session/session_controller.go:272","msg":"Add ref","type":"controller","Request.Namespace":"default","Request.Name":"jorge.martinez-wsoqj","name":"nginx"}
{"level":"info","ts":1619887918.6608238,"logger":"root","caller":"v2/klog.go:916","msg":"Throttling request took 1.045354551s, request: GET:https://10.43.0.1:443/apis/config.gatekeeper.sh/v1alpha1?timeout=32s\n"}
{"level":"error","ts":1619887919.4133584,"logger":"root","caller":"openshift/deploymentconfig.go:62","msg":"Could not get DeploymentConfig","type":"controller","Request.Namespace":"default","Request.Name":"jorge.martinez-wsoqj","name":"","error":"failed finding DeploymentConfig nginx in namespace default: no matches for kind \"DeploymentConfig\" in version \"apps.openshift.io/v1\"","errorVerbose":"no matches for kind \"DeploymentConfig\" in version \"apps.openshift.io/v1\"\nfailed finding DeploymentConfig nginx in namespace default\ngithub.com/maistra/istio-workspace/pkg/openshift.getDeploymentConfig\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/pkg/openshift/deploymentconfig.go:169\ngithub.com/maistra/istio-workspace/pkg/openshift.DeploymentConfigLocator\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/pkg/openshift/deploymentconfig.go:57\ngithub.com/maistra/istio-workspace/controllers/session.(*ReconcileSession).sync\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/controllers/session/session_controller.go:318\ngithub.com/maistra/istio-workspace/controllers/session.(*ReconcileSession).syncAllRefs\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/controllers/session/session_controller.go:274\ngithub.com/maistra/istio-workspace/controllers/session.(*ReconcileSession).Reconcile\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/controllers/session/session_controller.go:214\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:298\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:253\nsigs.k8s.io/controller-runtime/p
kg/internal/controller.(*Controller).Start.func1.2\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:216\nk8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext.func1\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:185\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:155\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:156\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133\nk8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:185\nk8s.io/apimachinery/pkg/util/wait.UntilWithContext\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:99\nruntime.goexit\n\t/snap/go/current/src/runtime/asm_amd64.s:1371","stacktrace":"github.com/maistra/istio-workspace/pkg/openshift.DeploymentConfigLocator\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/pkg/openshift/deploymentconfig.go:62\ngithub.com/maistra/istio-workspace/controllers/session.(*ReconcileSession).sync\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/controllers/session/session_controller.go:318\ngithub.com/maistra/istio-workspace/controllers/session.(*ReconcileSession).syncAllRefs\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/controllers/session/session_controller.go:274\ngithub.com/maistra/istio-workspace/controllers/session.(*ReconcileSession).Reconcile\n\t/home/circleci/.go_workspac
e/src/github.com/maistra/istio-workspace/controllers/session/session_controller.go:214\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:298\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:253\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1.2\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:216\nk8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext.func1\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:185\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:155\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:156\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133\nk8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:185\nk8s.io/apimachinery/pkg/util/wait.UntilWithContext\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:99"}
{"level":"info","ts":1619887919.5145895,"logger":"root","caller":"k8s/deployment.go:82","msg":"Found Deployment","type":"controller","Request.Namespace":"default","Request.Name":"jorge.martinez-wsoqj","name":"nginx"}
{"level":"error","ts":1619887919.5184596,"logger":"root","caller":"v2/klog.go:914","msg":"Observed a panic: \"invalid memory address or nil pointer dereference\" (runtime error: invalid memory address or nil pointer dereference)\ngoroutine 577 [running]:\nk8s.io/apimachinery/pkg/util/runtime.logPanic(0x1818520, 0x26ea430)\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:74 +0x95\nk8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:48 +0x86\npanic(0x1818520, 0x26ea430)\n\t/snap/go/current/src/runtime/panic.go:965 +0x1b9\ngithub.com/maistra/istio-workspace/pkg/istio.DestinationRuleMutator(0x1c3ef98, 0xc0007a9f80, 0xc0007f1d70, 0x14, 0xc00053d369, 0x7, 0xc00053d4c0, 0x6, 0xc00053d428, 0x8, ...)\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/pkg/istio/destinationrule.go:72 +0x645\ngithub.com/maistra/istio-workspace/controllers/session.(*ReconcileSession).sync(0xc0000bc550, 0x1c3ef98, 0xc0007a9f80, 0xc0007f1d70, 0x14, 0xc00053d369, 0x7, 0xc00053d4c0, 0x6, 0xc00053d428, ...)\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/controllers/session/session_controller.go:324 +0x305\ngithub.com/maistra/istio-workspace/controllers/session.(*ReconcileSession).syncAllRefs(0xc0000bc550, 0x1c3ef98, 0xc0007a9f80, 0xc0007f1d70, 0x14, 0xc00053d369, 0x7, 0xc00053d4c0, 0x6, 0xc00053d428, ...)\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/controllers/session/session_controller.go:274 +0xa30\ngithub.com/maistra/istio-workspace/controllers/session.(*ReconcileSession).Reconcile(0xc0000bc550, 0x1c3ef98, 0xc0007a9f80, 0xc00053d369, 0x7, 0xc0007f1d70, 0x14, 0xc0007a9f80, 0xc000030000, 0x18aaa00, ...)\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/controllers/session/session_controller.go:214 +0xac8\nsigs.k8s.
io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc00026ac80, 0x1c3eef0, 0xc0005b0000, 0x18704a0, 0xc000930a00)\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:298 +0x30d\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc00026ac80, 0x1c3eef0, 0xc0005b0000, 0x100680da87ff0800)\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:253 +0x205\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1.2(0x1c3eef0, 0xc0005b0000)\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:216 +0x4a\nk8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext.func1()\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:185 +0x37\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc0006ccf50)\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:155 +0x5f\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000963f50, 0x1c07a20, 0xc0007a9f20, 0xc0005b0001, 0xc000522540)\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:156 +0x9b\nk8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc0006ccf50, 0x3b9aca00, 0x0, 0x72656c6c69540601, 0xc000522540)\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x98\nk8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext(0x1c3eef0, 0xc0005b0000, 0xc0006786a0, 0x3b9aca00, 0x0, 0x762d6769666e6f01)\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:185 +0xa6\nk8s.io/apimachinery/pkg/util/w
ait.UntilWithContext(0x1c3eef0, 0xc0005b0000, 0xc0006786a0, 0x3b9aca00)\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:99 +0x57\ncreated by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:213 +0x40d\n","stacktrace":"k8s.io/klog/v2.(*loggingT).output\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/k8s.io/klog/v2/klog.go:914\nk8s.io/klog/v2.(*loggingT).printf\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/k8s.io/klog/v2/klog.go:751\nk8s.io/klog/v2.Errorf\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/k8s.io/klog/v2/klog.go:1463\nk8s.io/apimachinery/pkg/util/runtime.logPanic\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:78\nk8s.io/apimachinery/pkg/util/runtime.HandleCrash\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:48\nruntime.gopanic\n\t/snap/go/current/src/runtime/panic.go:965\nruntime.panicmem\n\t/snap/go/current/src/runtime/panic.go:212\nruntime.sigpanic\n\t/snap/go/current/src/runtime/signal_unix.go:734\ngithub.com/maistra/istio-workspace/pkg/istio.DestinationRuleMutator\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/pkg/istio/destinationrule.go:72\ngithub.com/maistra/istio-workspace/controllers/session.(*ReconcileSession).sync\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/controllers/session/session_controller.go:324\ngithub.com/maistra/istio-workspace/controllers/session.(*ReconcileSession).syncAllRefs\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/controllers/session/session_controller.go:274\ngithub.com/maistra/istio-workspace/controllers
/session.(*ReconcileSession).Reconcile\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/controllers/session/session_controller.go:214\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:298\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:253\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1.2\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:216\nk8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext.func1\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:185\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:155\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:156\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133\nk8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:185\nk8s.io/apimachinery/pkg/util/wait.UntilWithContext\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:99"}
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x168b585]
goroutine 577 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:55 +0x109
panic(0x1818520, 0x26ea430)
	/snap/go/current/src/runtime/panic.go:965 +0x1b9
github.com/maistra/istio-workspace/pkg/istio.DestinationRuleMutator(0x1c3ef98, 0xc0007a9f80, 0xc0007f1d70, 0x14, 0xc00053d369, 0x7, 0xc00053d4c0, 0x6, 0xc00053d428, 0x8, ...)
	/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/pkg/istio/destinationrule.go:72 +0x645
github.com/maistra/istio-workspace/controllers/session.(*ReconcileSession).sync(0xc0000bc550, 0x1c3ef98, 0xc0007a9f80, 0xc0007f1d70, 0x14, 0xc00053d369, 0x7, 0xc00053d4c0, 0x6, 0xc00053d428, ...)
	/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/controllers/session/session_controller.go:324 +0x305
github.com/maistra/istio-workspace/controllers/session.(*ReconcileSession).syncAllRefs(0xc0000bc550, 0x1c3ef98, 0xc0007a9f80, 0xc0007f1d70, 0x14, 0xc00053d369, 0x7, 0xc00053d4c0, 0x6, 0xc00053d428, ...)
	/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/controllers/session/session_controller.go:274 +0xa30
github.com/maistra/istio-workspace/controllers/session.(*ReconcileSession).Reconcile(0xc0000bc550, 0x1c3ef98, 0xc0007a9f80, 0xc00053d369, 0x7, 0xc0007f1d70, 0x14, 0xc0007a9f80, 0xc000030000, 0x18aaa00, ...)
	/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/controllers/session/session_controller.go:214 +0xac8
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc00026ac80, 0x1c3eef0, 0xc0005b0000, 0x18704a0, 0xc000930a00)
	/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:298 +0x30d
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc00026ac80, 0x1c3eef0, 0xc0005b0000, 0x100680da87ff0800)
	/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:253 +0x205
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1.2(0x1c3eef0, 0xc0005b0000)
	/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:216 +0x4a
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext.func1()
	/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:185 +0x37
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc0006ccf50)
	/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:155 +0x5f
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000963f50, 0x1c07a20, 0xc0007a9f20, 0xc0005b0001, 0xc000522540)
	/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:156 +0x9b
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc0006ccf50, 0x3b9aca00, 0x0, 0x72656c6c69540601, 0xc000522540)
	/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x98
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext(0x1c3eef0, 0xc0005b0000, 0xc0006786a0, 0x3b9aca00, 0x0, 0x762d6769666e6f01)
	/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:185 +0xa6
k8s.io/apimachinery/pkg/util/wait.UntilWithContext(0x1c3eef0, 0xc0005b0000, 0xc0006786a0, 0x3b9aca00)
	/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:99 +0x57
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
	/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:213 +0x40d`
</details>

After this in command line I have this error:
```
ike develop --deployment nginx --port 80 --watch --run 'docker run -it -p 80:80 nginx' --route header:end-user=alien-ike --watch
0.109
failed waiting for deployment to create	{"error": "timed out waiting for the condition"}
Error: failed executing develop command: no Deployment or DeploymentConfig found for target 'nginx'
failed executing command	{"error": "failed executing develop command: no Deployment or DeploymentConfig found for target 'nginx'", "errorVerbose": "no Deployment or DeploymentConfig found for target 'nginx'\nfailed executing develop command\ngithub.com/maistra/istio-workspace/pkg/cmd/develop.NewCmd.func2\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/pkg/cmd/develop/cmd.go:48\ngithub.com/spf13/cobra.(*Command).execute\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/github.com/spf13/cobra/command.go:852\ngithub.com/spf13/cobra.(*Command).ExecuteC\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/github.com/spf13/cobra/command.go:960\ngithub.com/spf13/cobra.(*Command).Execute\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/vendor/github.com/spf13/cobra/command.go:897\nmain.main\n\t/home/circleci/.go_workspace/src/github.com/maistra/istio-workspace/cmd/ike/main.go:47\nruntime.main\n\t/snap/go/current/src/runtime/proc.go:225\nruntime.goexit\n\t/snap/go/current/src/runtime/asm_amd64.s:1371"}
```

##### Additional Information

Anything relevant to help us resolve the problem. For example:

- K8S 1.17.17
- Istio 1.4.10
- Istio-workspace latest install via olm

### PR DESCRIPTION
**Pull request by aslakknutsen**
If the target subset is in a connected vs we need to both create
a new virtual service to auto add the header as well as manipulate
the service directly to match the header.

* remove test-service url path to simplify examples/test code
* log hostname.String()

Fixes #827

### CODE REVIEW

File: config/manager/kustomization.yaml [hunk 1]
```diff
@@ -12,5 +12,5 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 images:
 - name: controller
-  newName: quay.io/maistra/istio-workspace
+  newName: quay.io/maistra-dev/istio-workspace
```
Comment (aslakknutsen) [inline config/manager/kustomization.yaml:15]:
```suggestion
  newName: quay.io/maistra/istio-workspace
```

**Review by bartoszmajsak — APPROVED** [review]
IgFdKILY6TZyIJO3YU

**Comment by alien-ike** [comment]
@aslakknutsen: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
e2e-tests-openshift | e4b01fe90119de13dcdcd6fcf786b4bfe5e23bd3 | [link](http://status-prow.apps.workspace.maistra.test/view/gs/ike-prow-logs/pr-logs/pull/maistra_istio-workspace/838/e2e-tests-openshift/1392146676805799936) | `/test openshift`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Pull request merged**